src/woo_publications/publications/utils.py

```diff
@@ -0,0 +1,14 @@
+from uuid import UUID
+
+from django.http import HttpRequest
+from django.urls import reverse_lazy
+
+
+def absolute_document_download_uri(request: HttpRequest, document_uuid: UUID) -> str:
```

Comment:
I wouldn't make this a utility function, but instead a method on the `Document` model. As far as I can see, this is always being called from places where a `Document` instance is actually in scope. This function is also tightly coupled with the `Document` model, so I'd rather go for:

```py
from rest_framework.reverse import reverse

class Document(...):
    def build_absolute_api_download_uri(self, request: HttpRequest) -> str:
        return reverse("api:document_download", kwargs={"uuid": self.uuid}, request=request)
```

The DRF `reverse` version already handles making URLs fully qualified and it also supports any potential version negotiation from API endpoints (if we'll ever need that in the future).